src/api/nodeRpcCall.js

```diff
@@ -1924,6 +1924,63 @@ const fetchPendingIdentities = async () => {
   return processed.filter((entity) => entity.data.judgements.length === 0);
 };
 
+const getSignaturesForGivenId = async (allSignatures, id) => allSignatures.find(
+  ([contractId]) => contractId.args[0].eq(id),
+)[1].unwrapOrDefault();
+
+const getAllContracts = async () => {
+  const api = await getApi();
+
+  const [rawJudgesSignatures, rawPartiesSignatures] = await Promise.all([
+    api.query.contractsRegistry.judgesSignatures.entries(),
+    api.query.contractsRegistry.partiesSignatures.entries(),
+  ]);
+
+  const rawContracts = await api.query.contractsRegistry.contracts.entries();
+  return Promise.all(rawContracts.map(async ([id, contract]) => {
```

Comment:
There's nothing async here except for `getSignaturesForGivenId` which shouldn't be async.. So no need for `async` nor Promise.all here.